lib/sdm.js

```diff
@@ -31,7 +36,28 @@ module.exports = class SDMAttachmentsService extends (
     return this.creds;
   }
 
+  async checkRepositoryType(req) {
+    const { repositoryId } = getConfigurations();
+    //check if repository is versionable
+    let repotype = cache.get(repositoryId) 
+    let isVersioned;
+    if (repotype == undefined) {
+      const token = await fetchAccessToken(
```

Comment:
this wont work because we need to have client credentials flow not user token flow. Just deploy to BTP and check once

Reply:
Done. Added client Credentials Flow for getRepoInfo